### PR TITLE
Stats: hide sensible data

### PR DESCRIPTION
**Pull request by luckydonald**
Adds the `--stats-hide-sensible-data` flag (Docker: `$TELEGRAM_STAT_HIDE_SENSIBLE_DATA=1`),
Removing the bot token and the Webhook url from the stats endpoint.

### CODE REVIEW

**Comment by MarcoBuster** [comment]
I'm confused: why did you created another pull request cherry picking a feature from #22?

**Comment by luckydonald** [comment]
To enable this to be merged already while we still try to figure out the other problems.

File: telegram-bot-api/ClientManager.cpp [hunk 1]
```diff
@@ -289,9 +290,17 @@ void ClientManager::get_stats(td::PromiseActor<td::BufferSlice> promise,
...
     sb << "id\t" << bot_info.id_ << "\n";
     sb << "uptime\t" << now - bot_info.start_time_ << "\n";
-    sb << "token\t" << bot_info.token_ << "\n";
+    if (!parameters_->stats_hide_sensible_data_) {
+      sb << "token\t" << bot_info.token_ << "\n";
+    }
     sb << "username\t" << bot_info.username_ << "\n";
-    sb << "webhook\t" << bot_info.webhook_ << "\n";
+    if (!parameters_->stats_hide_sensible_data_) {
+        sb << "webhook\t" << bot_info.webhook_ << "\n";
```
Comment (luckydonald) [inline telegram-bot-api/ClientManager.cpp:298]: Ooops a space too much here, but that would be cleaned up with the json one, #23

Comment (MarcoBuster) [reply telegram-bot-api/ClientManager.cpp:298]: You can fix it here then rebase the another branch with upstream when this it's merged

**Comment by luckydonald** [comment]
I didn't see your comment, that two spaces is fixed now.

File: README.md [hunk 2]
```diff
@@ -97,6 +97,9 @@ Allow http connection in non-local mode
 #### Executable parameter `max-batch-operations`
 maximum number of batch operations (default 10000)
 
+##### Executable parameter `--hide-sensible-data`
```
Comment (MarcoBuster) [inline README.md:100]: It's `--stats-hide-sensible-data`

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -19,6 +19,9 @@ CUSTOM_ARGS=""
 if [ -n "$TELEGRAM_STAT" ]; then
   CUSTOM_ARGS="${CUSTOM_ARGS} --http-stat-port=8082"
 fi
+if [ -n "$TELEGRAM_STAT_HIDE_SENSIBLE_DATA" ]; then
```
Comment (MarcoBuster) [inline docker-entrypoint.sh:22]: "stats" or "stat"?

File: telegram-bot-api/HttpStatConnection.cpp [hunk 4]
```diff
@@ -23,7 +23,7 @@ void HttpStatConnection::handle(td::unique_ptr<td::HttpQuery> http_query,
   init_promise_future(&promise, &future);
   future.set_event(td::EventCreator::yield(actor_id()));
   LOG(DEBUG) << "SEND";
-  send_closure(client_manager_, &ClientManager::get_stats, std::move(promise), http_query->get_args());
+  send_closure(client_manager_, &ClientManager::get_stats, std::move(promise), http_query->get_args(), true);
```
Comment (MarcoBuster) [inline telegram-bot-api/HttpStatConnection.cpp:26]: What's that?

File: telegram-bot-api/ClientManager.h [hunk 5]
```diff
@@ -46,7 +46,7 @@ class ClientManager final : public td::Actor {
 
   bool check_flood_limits(PromisedQueryPtr &query, bool is_user_login=false);
 
-  void get_stats(td::PromiseActor<td::BufferSlice> promise, td::vector<std::pair<td::string, td::string>> args);
+  void get_stats(td::PromiseActor<td::BufferSlice> promise, td::vector<std::pair<td::string, td::string>> args, bool as_json);
```
Comment (MarcoBuster) [inline telegram-bot-api/ClientManager.h:49]: Out of scope of this PR

File: telegram-bot-api/ClientManager.cpp [hunk 6]
```diff
@@ -202,7 +202,8 @@ bool ClientManager::check_flood_limits(PromisedQueryPtr &query, bool is_user_log
 }
 
 void ClientManager::get_stats(td::PromiseActor<td::BufferSlice> promise,
-                              td::vector<std::pair<td::string, td::string>> args) {
```
Comment (MarcoBuster) [inline telegram-bot-api/ClientManager.cpp:205]: Out of scope of this PR

**Review by MarcoBuster — CHANGES_REQUESTED** [review]

File: telegram-bot-api/HttpStatConnection.cpp (hunk 4 above)
Comment (luckydonald) [reply telegram-bot-api/HttpStatConnection.cpp:26]: Out of scope of this PR

File: docker-entrypoint.sh (hunk 3 above)
Comment (luckydonald) [reply docker-entrypoint.sh:22]: I'd say stats, but the other variable already is `$TELEGRAM_STAT`, not stats.

**Comment by luckydonald** [comment]
So far did what you asked me to change.

File: docker-entrypoint.sh (hunk 3 above)
Comment (MarcoBuster) [reply docker-entrypoint.sh:22]: Uh, ok

**Review by MarcoBuster — APPROVED** [review]

**Comment by MarcoBuster** [comment]
LGTM. Please resolve merge conflicts so we can merge as soon @giuseppeM99 and @andrew-ld approve

**Comment by luckydonald** [comment]
Regarding 
> `$TELEGRAM_STAT_HIDE_SENSIBLE_DATA` or 
> `$TELEGRAM_STATS_HIDE_SENSIBLE_DATA`?

@MarcoBuster https://t.me/TDLightChat/1475
> use stat. in the future, we'll rewrite the docs and migrate to stats

**Review by giuseppeM99 — CHANGES_REQUESTED** [review]
I see that we've agreed to use `stat` instead of `stats`, change the content of the code so that we can merge

**Comment by luckydonald** [comment]
Ah, that was for the environment variable?

**Review by andrew-ld — APPROVED** [review]

**Review by giuseppeM99 — APPROVED** [review]

**Comment by luckydonald** [comment]
Great, it's approved by everyone now.
@MarcoBuster do you need anything else for a merge?

**Comment by giuseppeM99** [comment]
Should we merge this or just wait for #22 to be ready?

**Comment by luckydonald** [comment]
I'd rather have that ready as soon as possible.

**Comment by luckydonald** [comment]
As you from what I've seen don't want to take an iterative approach with #22 it'll take a while till we get those flag stuff included there to be merged.
The branch there is working fine and thus could be merged, but your decision seems to be to wait until we got the exact json data worked out completely.

I have probably no time to work on it this week.

**Comment by MarcoBuster** [comment]
3 approvals, checks passed, tests passed. No reasons to wait.

**Pull request merged**